### ISSUE TITLE
[v3] useQuery with suspense doesn't use cache data on mount

### ISSUE THREAD

**Issue by s-belous** [tannerlinsley/react-query#1249]
**Describe the bug**
When the `useQuery` is being unmounted and mounted again, it throws suspense even if cacheTime wasn't expired.

**To Reproduce**
1. Go to https://codesandbox.io/s/nice-firefly-b50jc?file=/src/App.js
2. Tap on 'Hide time' button
3. Wait for 2 seconds (default staleTime)
4. Tap on 'Show time' button
5. Loading indicator appeared instead of a previously cached result

**Expected behavior**
I'd expect the behavior described in the documentation:
```
The query immediately returns the available cached value while the fetchTodos function is being run in the background to populate the query with a fresh value.
```

**Comment by tannerlinsley** [comment]
Tracking this in #1160 and #1059

**Comment by boschni** [comment]
Thanks for reporting! Fixed in `3.2.0-beta.21`
